lib/archive/unpack_test.go

```diff
@@ -0,0 +1,1035 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package archive
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware/vic/pkg/trace"
+)
+
+var (
+	// L1 directories
+	path1dir1 = "/path1dir1"
+	path2dir1 = "/path2dir1"
+
+	// l1 files
+	l1file1 = "/file1"
+	l1file2 = "/file2"
+
+	// l2 directories
+	path1dir2 = filepath.Join(path1dir1, "path1dir2")
+	path1dir3 = filepath.Join(path1dir1, "path1dir3")
+	path2dir2 = filepath.Join(path2dir1, "path2dir2")
+	path2dir3 = filepath.Join(path2dir1, "path2dir3")
+
+	// l2 files
+	l2file1 = filepath.Join(path1dir1, "file1")
+	l2file2 = filepath.Join(path1dir1, "file2")
+	l2file3 = filepath.Join(path2dir1, "file1")
+	l2file4 = filepath.Join(path2dir1, "file2")
+
+	// l3 directories
+	path1dir4 = filepath.Join(path1dir3, "path1dir4")
+
+	// l3 files
+	l3file1 = filepath.Join(path1dir3, "file1")
+	l3file2 = filepath.Join(path2dir2, "file1")
+)
+
```

Comment:
There is handling for that in the `unpack` function. I have added logging and a comment to that section. As well as including a simple invalid stream unit test. I could also devise a truncated tar stream test as well if you think it would add even more value.